cmsis-core/cmsis_nvic.h

```diff
@@ -34,6 +34,12 @@
 
 #include "cmsis.h"
 
+/* throw meaningful compiler error if yotta configuration is not provided */
+#if !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_OFFSET)      || !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_NUMBER) || \
```

Comment:
I see. But in reality, most of the `cmsis.h` files only include `cmsis_nvic.h`, not the whole deal. Conceptually I would expect that a file called `cmsis.h` includes *all* cmsis-related header files (which with the amenities of modular programming will end up being just the ones needed for the given platform).

We do that in [this header file](https://github.com/ARMmbed/uvisor-lib/blob/master/uvisor-lib/uvisor-lib.h) in uVisor, so that you can include only `uvisor-lib/uvisor-lib.h` and you get the complete package. So what it would probably make sense is to have `cmsis.h` always include `cmsis_generich.h` (which defines the symbols above, and would include `cmsis_nvic.h`) *plus* the hardware-specific stuff (like the platform main header file).

If we like this proposal the changes would be very simple and backward compatible (including `cmsis_generic.h` in `cmsis.h` would bring along also the files that were included individually).